File: src/main/java/no/nav/bidrag/beregn/nettobarnetilsyn/NettoBarnetilsynCore.java
package no.nav.bidrag.beregn.nettobarnetilsyn;

import no.nav.bidrag.beregn.nettobarnetilsyn.dto.BeregnNettoBarnetilsynGrunnlagCore;
import no.nav.bidrag.beregn.nettobarnetilsyn.dto.BeregnetNettoBarnetilsynResultatCore;
import no.nav.bidrag.beregn.nettobarnetilsyn.periode.NettoBarnetilsynPeriode;

public interface NettoBarnetilsynCore {

  BeregnetNettoBarnetilsynResultatCore beregnNettoBarnetilsyn(BeregnNettoBarnetilsynGrunnlagCore beregnNettoBarnetilsynGrunnlagCore);

  static NettoBarnetilsynCore getInstance() {
    return new NettoBarnetilsynCoreImpl(NettoBarnetilsynPeriode.getInstance());
  }
}

File: src/main/kotlin/no/nav/bidrag/beregn/nettobarnetilsyn/NettoBarnetilsynCore.kt
package no.nav.bidrag.beregn.nettobarnetilsyn

import no.nav.bidrag.beregn.nettobarnetilsyn.dto.BeregnNettoBarnetilsynGrunnlagCore
import no.nav.bidrag.beregn.nettobarnetilsyn.dto.BeregnetNettoBarnetilsynResultatCore
import no.nav.bidrag.beregn.nettobarnetilsyn.periode.NettoBarnetilsynPeriode

fun interface NettoBarnetilsynCore {
    fun beregnNettoBarnetilsyn(grunnlag: BeregnNettoBarnetilsynGrunnlagCore): BeregnetNettoBarnetilsynResultatCore

    companion object {
        fun getInstance(): NettoBarnetilsynCore = NettoBarnetilsynCoreImpl(NettoBarnetilsynPeriode.getInstance())
    }
}
